org.adempiere.base/src/org/compiere/tools/FileUtil.java

```diff
@@ -431,11 +436,77 @@ public static String getTempMailName(String subject, String extension) {
 		SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmss");
 		String dt = sdf.format(cal.getTime());
 		String cleanName = subject.replaceAll("[ &/]", "");
-		String localFile = System.getProperty("java.io.tmpdir")
-				+ System.getProperty("file.separator") + cleanName + "_" + dt + "_" + Env.getContext(Env.getCtx(), Env.AD_SESSION_ID)
-				+ extension;
+		String dir = System.getProperty("java.io.tmpdir")
+				+ System.getProperty("file.separator") + "rpttmp_" + dt + "_" + Env.getContext(Env.getCtx(), Env.AD_SESSION_ID) + System.getProperty("file.separator");
+		String localFile = dir 
+				+ cleanName	+ extension;
+		new File(dir ).mkdirs();
 		return localFile;
 	}
+	
+	/**
+	 * 
+	 * @param title
+	 * @param table_ID
+	 * @param record_ID
+	 * @param ctx
+	 * @param m_WindowNo
+	 * @param trxName
+	 * @return
+	 */
+	public static String parseTitle(Properties ctx, String title, int table_ID, int record_ID, int m_WindowNo, String trxName) {
+		if (title.contains("@") && record_ID>0 && table_ID>0) {
+			MTable table=new MTable(ctx,table_ID,trxName );
+			PO po = new GenericPO(table.getTableName(), ctx, record_ID, trxName);
+			title=Env.parseVariable(title, po, trxName, true);
+		}
+		else 
+			title=  Env.parseContext (ctx, m_WindowNo, title,
+					true, true);
+		return title;
+	}
+	
+    public static File createTempFile(String prefix, String suffix, File directory) throws IOException
+	{
+        if (prefix.length() < 3) {
+            throw new IllegalArgumentException("Prefix string \"" + prefix +
+                "\" too short: length must be at least 3");
+        }
+        if (suffix == null)
+            suffix = ".tmp";

```

Comment:
added condition and validation for prefix and suffix - yet another change of existing behavior that's not mention in the ticket.